ui/components/ui/sender-to-recipient/sender-to-recipient.component.js

```diff
@@ -124,6 +129,24 @@ export function RecipientWithAddress({
     );
   }
 
+  let displayName;
+  if (petnamesEnabled) {
+    displayName = (
+      <Name value={recipientAddress} type={NameType.ETHEREUM_ADDRESS} />
+    );
+  } else {
+    displayName =
+      recipientName ||
+      recipientNickname ||
+      recipientMetadataName ||
+      recipientEns ||
+      shortenAddress(checksummedRecipientAddress);
+
+    if (addressOnly && !displayName) {
```

Comment:
Looking at the original logic, is the `newContract` fallback used only when `addressOnly` is `false`?

Reply:
Good catch! Done in #22515